qa/lucene-index-compatibility/src/javaRestTest/java/org/elasticsearch/lucene/FullClusterRestartSystemIndexCompatibilityIT.java

```diff
@@ -83,19 +89,29 @@ public void testAsyncSearchIndexMigration() throws Exception {
                 ObjectPath.createFromResponse(client().performRequest(migrateRequest)).evaluate("features.0.feature_name"),
                 equalTo("async_search")
             );
-            assertBusy(() -> {
-                Request checkMigrateProgress = new Request("GET", "/_migration/system_features");
-                Response resp = null;
-                try {
-                    assertFalse(
-                        ObjectPath.createFromResponse(client().performRequest(checkMigrateProgress))
-                            .evaluate("migration_status")
-                            .equals("IN_PROGRESS")
-                    );
-                } catch (IOException e) {
-                    throw new AssertionError("System feature migration failed", e);
-                }
-            });
+
+            logger.debug("--> starting system index migration");
+            Request checkMigrateProgress = new Request("GET", "/_migration/system_features");
+            try {
+                assertBusy(() -> {
+                    Response resp = null;
+                    try {
+                        assertFalse(
+                            ObjectPath.createFromResponse(client().performRequest(checkMigrateProgress))
+                                .evaluate("migration_status")
+                                .equals("IN_PROGRESS")
+                        );
+                    } catch (IOException e) {
+                        throw new AssertionError("System feature migration failed", e);
+                    }
+                });
+            } catch (AssertionError e) {
+                logger.debug(
+                    "--> system index migration not finished yet, response: {}",
+                    EntityUtils.toString(client().performRequest(checkMigrateProgress).getEntity())
```

Comment:
Does this tell us much though when we execute it after the busy wait ran? Maybe instead we should/could make the responses part of the assertion error we accumulate?
They should be tiny anyway?